### DISCUSSION TITLE
Method Not Allowed (GET): /admin/logout/

### DISCUSSION THREAD

**Discussion by adhibtaAtFb** [stephenmcd/mezzanine#2071] [Q&A] [answered]
when I will log out via the logout button on the top right,
logout was unsuccessful with output:
![image](https://github.com/stephenmcd/mezzanine/assets/96231952/34017f51-42b7-4803-8cfc-3ba312b28b70)

but for the login button there is no problem..
thanks

**Comment by molokov** [comment, marked as answer]
In Django 5.0, they removed the ability to log out via a simple link (i.e. the GET method), see https://docs.djangoproject.com/en/5.0/releases/5.0/#features-removed-in-5-0

A solution is to change the logout link to a form in your template, for example:

```
<form action="{% url 'logout' %}" method="post">
  {% csrf_token %}
  <button type="submit">Log Out</button>
</form>
```

Or, if you still want it formatted as an anchor link (assumes jQuery):

```
<a href="#" onclick="$('#logoutform').submit()">Log Out</a>
<form id="logoutform" method="post" action="{% url 'logout' %}">
        {% csrf_token %}
        <input type="hidden">
 </form>
 ```
 
Given that Django 5.0 support in Mezzanine isn't official as yet, the templates haven't been updated - but that's certainly something we should consider adding. For now, update your own templates so the logout link uses a form like the above.

EDIT: I now see that you are commenting on logging out via the Admin interface. That is probably another template that should be updated. It's not as vital as the public log out link, however.

**Reply by adhibtaAtFb** [reply]
Thank you Molokov for your response,
first you said django 5 doesn't support GET, so I tried installing with django 4.2 and it went smoothly.
![image](https://github.com/stephenmcd/mezzanine/assets/96231952/65f850d8-a61a-4690-902d-16386881ba8a)
![image](https://github.com/stephenmcd/mezzanine/assets/96231952/230612ac-4c89-4da2-8f4d-75f003e16a89)


I don't know how to edit the mezzanine template, because I haven't studied the Django framework.
Thanks

**Reply by Parthian** [reply]
It's grappelli-safe that is the problem. See my answer below.

**Comment by Parthian** [comment]
After many hours I'll never get back I eventually found the Logout code is in grappelli, not Django nor Mezzanine.
env/lib/python3.10/site-packages/grappelli_safe/templates/admin/includes_grappelli/header.html
End the file with something like this, thanks chatgpt. Ideally I should override this template so my edit doesn't get killed on an update. Not sure how to do that.
```
        {% url "admin:logout" as logout_url %}
        {% if logout_url %}
        <li style="padding-top: 0;">
            <form style="margin: 0;" action="{{ logout_url }}" method="post">
                {% csrf_token %}
                <button style="background: transparent;" type="submit">{% trans 'Log out' %}</button>
            </form>
        </li>
        {% else %}
        <li style="padding-top: 0;">
            <form style="margin: 0;" action="{{ root_path }}logout/" method="post">
                {% csrf_token %}
                <button style="background: transparent;" type="submit">{% trans 'Log out' %}</button>
            </form>
        </li>
        {% endif %}
        {% endblock %}
    </ul>
    {% endif %}
</div>
```
